code/modules/projectiles/guns/misc.dm

```diff
@@ -107,6 +108,56 @@
 	recoil = RECOIL_AMOUNT_TIER_5
 	empty_sound = 'sound/weapons/gun_empty.ogg'
 
+/obj/item/weapon/gun/m60/clicked(mob/user, list/mods)
+	if(mods["alt"])
+		if(!CAN_PICKUP(user, src))
+			return ..()
+		if(!locate(src) in list(user.get_active_hand(), user.get_inactive_hand()))
+			return TRUE
+		if(user.get_active_hand() && user.get_inactive_hand())
+			to_chat(user, SPAN_WARNING("You can't do that with your hands full!"))
+			return TRUE
+		if(!cover_open)
+			playsound(src.loc, 'sound/handling/smartgun_open.ogg', 50, TRUE, 3)
+			to_chat(user, SPAN_NOTICE("You open \the [src]'s feed cover, allowing the belt to be removed."))
+			cover_open = TRUE
+		else
+			playsound(src.loc, 'sound/handling/smartgun_close.ogg', 50, TRUE, 3)
+			to_chat(user, SPAN_NOTICE("You close \the [src]'s feed cover."))
+			cover_open = FALSE
+		update_icon()
+		return TRUE
+	else
+		return ..()
+
+/obj/item/weapon/gun/m60/replace_magazine(mob/user, obj/item/ammo_magazine/magazine)
+	if(!cover_open)
+		to_chat(user, SPAN_WARNING("\The [src]'s feed cover is closed! You can't put a new belt in! (alt-click to open it)"))
+		return
+	. = ..()
+
+/obj/item/weapon/gun/m60/unload(mob/user, reload_override, drop_override, loc_override)
+	if(!cover_open)
+		to_chat(user, SPAN_WARNING("\The [src]'s feed cover is closed! You can't take out the belt! (alt-click to open it)"))
+		return
+	. = ..()
+
+/obj/item/weapon/gun/m60/update_icon()
+	. = ..()
+	if(cover_open)
+		overlays += "+[base_gun_icon]_cover_open"
+	else
+		overlays += "+[base_gun_icon]_cover_closed"
+
+/obj/item/weapon/gun/m60/able_to_fire(mob/living/user)
+	. = ..()
+	if(.)
+		var/mob/living/carbon/human/H = user
```

Comment:
why do you typecast this

Reply:
I did not write this code, I believe it was stan albatross who did this, I just adapted it for M60.